### PR TITLE
Feat/88 duplicates

### PR DESCRIPTION
**Pull request by f-lukas**
### Updates
- Option to decide if column with duplicate rows is added
- Option to ignore spaces during duplicate detection
- Option to specify a string that is used for duplicate detection

### CODE REVIEW

**Comment by f-lukas** [comment]
@arunge could you please check if the new features work as expected? Thanks!

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,12 @@
 # MpiIsoApp development version
 
+## Version 23.02.3
+
+### Updates
+- Option to decide if column with duplicate rows is added
+- Option to ignore spaces during duplicate detection
```
Comment (arunge) [inline NEWS.md:7]: Could there be a bug here?

Without "Ignore Spaces" (1):
image

With "Ignore Spaces" (2):
image

File: R/01-createDuplicateModal.R [hunk 2]
```diff
@@ -27,7 +27,8 @@ createDuplicateModal <- function(vars, session) {
                 A value of 0 means that the values are rounded to whole numbers.
                 You can also enter negative numbers, which will result in rounding to the power of ten, e.g. the value -2 rounds to the nearest hundred.
                 For text columns you can choose between 'Case Sensitive', where a distinction is made between upper and lower case letters and
-                'Case Insensitive', where no distinction is made.
+                'Case Insensitive', where no distinction is made. Furthermore you have the option to ignore empty cells and spaces.
+                Optional you can specify a specific string. If you do so, rows that contain this string in the selected column are marked as duplicate.
```
Comment (arunge) [inline R/01-createDuplicateModal.R:31]:
```suggestion
                Optionally, you can specify a specific string. If you do so, rows that contain this string in the selected column are marked as duplicate.
```

File: R/01-createDuplicateModal.R [hunk 3]
```diff
@@ -27,7 +27,8 @@ createDuplicateModal <- function(vars, session) {
                 A value of 0 means that the values are rounded to whole numbers.
                 You can also enter negative numbers, which will result in rounding to the power of ten, e.g. the value -2 rounds to the nearest hundred.
```
Comment (arunge) [inline R/01-createDuplicateModal.R:28]:
```suggestion
                You can also enter negative numbers, which will result in rounding to the power of ten, e.g. the value -2 rounds to the nearest hundred.<br><br>
```

**Review by arunge — CHANGES_REQUESTED** [review]
Please check the white spaces option, the other two new options are working. :+1:

**Comment by f-lukas** [comment]
Very good that you noticed it. The combination of specifying a string and ignoring spaces didn't work because I only removed the spaces in the data and not in the specified string. It should now work.

**Review by arunge — COMMENTED** [review]
The issue with the white spaces is fixed now. :+1: 

There is something else: In the sidebar one can specify the columns to be displayed within the app:

image

In the pop up there are always all columns shown:

image

Maybe this is even a feature:
- there are less reactive dependencies between the sidebar selection and the duplicate search
- for me it does not to be a disadvantage to display the full data within the duplicate search. 

So we can even keep this as it is now. What do you think?

However, there seems to be an issue with the cutting of strings, it is not working always:

image

Can you check this? If this is a general issue and we should adjust the function in DataTools, please let me know.

Possibly this has to do with: https://github.com/Pandora-IsoMemo/iso-app/pull/101#issuecomment-1429941764
I am checking now.

**Comment by f-lukas** [comment]
The reason for this is the same as in https://github.com/Pandora-IsoMemo/iso-app/pull/101#issuecomment-1429941764, where I just implemented a workaround. It is not a bug but `cutAllLongStrings` only cuts character columns but not factor. There are quite many factor columns in the IsoMemo data sets. Maybe we can add the option to convert factor to character before cutting in `cutAllLongStrings`. What do you think?

**Review by arunge — APPROVED** [review]
Regarding:

> Can you check this? If this is a general issue and we should adjust the function in DataTools, please let me know.

We will solve this problem within another issue. Thus this PR is approved.

**Pull request merged**